test/unit-test/core_mqtt_utest.c

```diff
@@ -853,7 +867,7 @@ void test_MQTT_Connect_partial_receive()
 
     setupTransportInterface( &transport );
     setupNetworkBuffer( &networkBuffer );
-    transport.recv = transportRecvOneByte;
+    transport.recv = transportRecvNoData;
```

Comment:
Is it possible to write a test to replicate the scenario which caught this issue:

-  revc returns less than requested data (possibly return n/2 bytes so that second call with return complete data). 
- processLoop is called with 0 timeout. 

This test should fail without this change and pass with this change.

Reply:
We can achieve that with the existing mock recv function that returns only byte at a time. 
As the bug being addressed is primarily with "_less than expected data received_" when the "remaining timeout" value is nearing zero, I can add a test to mimic that by passing `0` as the timeout to `MQTT_ProcessLoop` and using the `transportRecvOneByte` mock function

Reply:
Added the test